File: benchmarks/CWE-330-WeakRandom/weak-random.py
# Vulnerable: random module for security purposes
import random
import string

def generate_token():
    # Vulnerable: random module is not cryptographically secure
    return ''.join(random.choices(string.ascii_letters + string.digits, k=32))

def generate_session_id():
    # Vulnerable: predictable
    return 'sess_' + str(random.random())[2:18]

def generate_reset_code():
    # Vulnerable: weak random for security code
    return str(random.randint(1000, 9999))

File: benchmarks/CWE-330-WeakRandom/weak-random.js
// Vulnerable: Math.random() for security purposes
function generateToken() {
  // Vulnerable: Math.random() is not cryptographically secure
  return Math.random().toString(36).substring(2);
}

function generateSessionId() {
  // Vulnerable: predictable session ID
  return 'sess_' + Math.random().toString(36).substr(2, 16);
}

function generateResetCode() {
  // Vulnerable: 4-digit code from Math.random
  return Math.floor(1000 + Math.random() * 9000).toString();
}
